okonomiyaki/file_formats/_egg_info.py

```diff
@@ -0,0 +1,697 @@
+
+
+Dependencies = collections.namedtuple("Dependencies", ["build", "runtime"])
+
+_TAG_RE = re.compile("(?P<interpreter>(cp|pp|cpython))(?P<version>([\d_]+))")
+
+
+def _python_tag_to_python(python_tag):
+    # This converts only python version we currently intent to support in
+    # metadata version 1.x.
+    if python_tag is None:
+        return None
+
+    generic_msg = "Python tag {0!r} not understood".format(python_tag)
+
+    m = _TAG_RE.match(python_tag)
+    if m is None:
+        raise InvalidMetadata(generic_msg)
+    else:
+        d = m.groupdict()
+        version = d["version"]
+        if len(version) == 1:
+            msg = "Version {0!r} not supported".format(version)
+            raise InvalidMetadata(msg)
+        elif len(version) == 2:
+            return "{0}.{1}".format(version[0], version[1])
+        else:
+            raise InvalidMetadata(generic_msg)
+
+
+def _metadata_version_to_tuple(metadata_version):
+    """ Convert a metadata version string to a tuple for comparison."""
+    return tuple(int(s) for s in metadata_version.split("."))
+
+
+def _normalized_info_from_string(spec_depend_string):
+    """ Return a 'normalized' dictionary from the given spec/depend string.
+
+    Note: the name value is NOT lower-cased, so that the egg filename may
+    rebuilt from the data.
+    """
+    raw_data = parse_rawspec(spec_depend_string)
+
+    data = {}
+    for k in (_TAG_METADATA_VERSION,
+              _TAG_NAME, _TAG_VERSION, _TAG_BUILD,
+              _TAG_ARCH, _TAG_OSDIST, _TAG_PLATFORM,
+              _TAG_PYTHON, _TAG_PACKAGES):
+        data[k] = raw_data[k]
+
+    metadata_version_info = _metadata_version_to_tuple(
+        data[_TAG_METADATA_VERSION]
+    )
+    if metadata_version_info[:2] < (1, 2):
+        data[_TAG_PYTHON_TAG] = _get_default_python_tag(raw_data,
+                                                        raw_data[_TAG_PYTHON])
+    else:
+        data[_TAG_PYTHON_TAG] = raw_data[_TAG_PYTHON_TAG]
+
+    return data
+
+
+class EggMetadata(object):
+    """ Enthought egg metadata for format 1.x.
+    """
+    @classmethod
+    def from_egg(cls, path_or_file):
+        """ Create a EggMetadata instance from an existing Enthought egg.
+
+        Parameters
+        ----------
+        path: str or file-like object.
+            If a string, understood as the path to the egg. Otherwise,
+            understood as a zipfile-like object.
+        """
+        def _read_summary(fp):
+            summary_arcname = "EGG-INFO/spec/summary"
+            try:
+                summary = fp.read(summary_arcname)
+            except KeyError:
+                # the summary file may not exist for eggs built with
+                # endist/repack
+                summary = b""
+            return summary.decode()
+
+        if isinstance(path_or_file, six.string_types):
+            spec_depend = LegacySpecDepend.from_egg(path_or_file)
+            with zipfile2.ZipFile(path_or_file) as fp:
```

Comment:
I would like to avoid putting an arbitrary limit on the #files inside an egg.

Maybe we could tweak LeanZipFile API to understand `max_files_count = None` as no limit (and use that for a default) ?